packages/dashboard/src/components/alert-store.tsx

```diff
@@ -1,77 +1,361 @@
+          />
+          {(alertRequest.message.length > 0 || additionalAlertMessage !== null) && (
+            <TextField
+              label="Message"
+              id="standard-size-small"
+              size="small"
+              variant="filled"
+              sx={{
+                '& .MuiFilledInput-root': {
+                  fontSize: isScreenHeightLessThan800 ? '0.8rem' : '1.15',
+                },
+              }}
+              InputLabelProps={{ style: { fontSize: isScreenHeightLessThan800 ? 16 : 20 } }}
+              InputProps={{ readOnly: true, className: classes.textField }}
+              fullWidth={true}
+              multiline
+              maxRows={4}
+              margin="dense"
+              value={
+                (alertRequest.message.length > 0 ? alertRequest.message : 'n/a') +
+                '\n' +
+                (additionalAlertMessage ?? '')
+              }
+            />
+          )}
+        </DialogContent>
+        <DialogActions>
+          {alertRequest.responses_available.map((response) => {
+            return (
+              <Button
+                size="small"
+                variant="contained"
+                autoFocus
+                key={`${alertRequest.id}-${response}`}
+                sx={{
+                  fontSize: isScreenHeightLessThan800 ? '0.8rem' : '1rem',
+                  padding: isScreenHeightLessThan800 ? '4px 8px' : '6px 12px',
+                }}
+                onClick={async () => {
+                  await respondToAlert(alertRequest.id, response);
+                  AppEvents.refreshAlert.next();
+                  setIsOpen(false);
+                }}
+              >
+                {response}
+              </Button>
+            );
+          })}
+          {alertRequest.task_id ? (
+            <TaskCancelButton
+              taskId={alertRequest.task_id}
+              size="small"
+              variant="contained"
+              color="secondary"
+              buttonText={'Cancel task'}
+              sx={{
+                fontSize: isScreenHeightLessThan800 ? '0.8rem' : '1rem',
+                padding: isScreenHeightLessThan800 ? '4px 8px' : '6px 12px',
+              }}
+            />
+          ) : null}
+          <Button
+            size="small"
+            variant="contained"
+            autoFocus
+            sx={{
+              fontSize: isScreenHeightLessThan800 ? '0.8rem' : '1rem',
+              padding: isScreenHeightLessThan800 ? '4px 8px' : '6px 12px',
+            }}
+            onClick={() => {
+              onDismiss();
+              setIsOpen(false);
+            }}
+          >
+            Dismiss
+          </Button>
+        </DialogActions>
+      </Dialog>
+    </>
+  );
+});
+
+export const AlertStore = React.memo(() => {
+  const rmf = React.useContext(RmfAppContext);
+  const [openAlerts, setOpenAlerts] = React.useState<Record<string, AlertRequest>>({});
```

Comment:
we already do that in https://github.com/open-rmf/rmf-web/pull/948/files#diff-1eb38a21aba250063703669fba901299924aa505db5b308230616fcdb3eedbbfR314

Reply:
The obs should be piped together and we do only a single subscription. e.g.

https://github.com/open-rmf/rmf-web/blob/2bfd48485b39ce8fecb75c3e691855957c0d2d4f/packages/dashboard/src/components/alert-store.tsx#L307-L320

both `rmf.alertRequestsObsStore` and `AppEvents.alertListOpenedAlert` affects the list of opened alerts, it leds to unnecessary complexity to avoid double refresh etc. We should do something like `rmf.alertRequestsObsStore.pipe((alert) => checkIfShouldOpenDialog(alert) && AppEvents.alertListOpenedAlert.next(alert))`, to keep a single source of truth. Also, `AppEvents.alertListOpenedAlert` should hold the canonical truth of the list of opened alert, i.e. we should filter before emitting onto it.

Reply:
https://github.com/open-rmf/rmf-web/pull/948/commits/fa1ba7e23be0e5bff97305021f86309c31223612